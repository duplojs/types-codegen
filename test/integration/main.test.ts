import { spawnSync } from "child_process";
import { existsSync, readFileSync, unlinkSync } from "fs";

describe("command", () => {
	const outputPath = `${process.cwd()}/test/output/outputTypeCodegen.d.ts`;

	beforeAll(() => {
		if (existsSync(outputPath)) {
			unlinkSync(outputPath);
		}
	});

	it("generate file", () => {
		spawnSync(
			"npx",
			["duplojs-types-codegen", "-i", "routes", "-o", "../output/outputTypeCodegen.d.ts"],
			{
				cwd: `${process.cwd()}/test/integration`,
			},
		);

		expect(existsSync(outputPath))
			.toBe(true);

		expect(readFileSync(outputPath, "utf-8"))
			.toMatchSnapshot();
	});
});
