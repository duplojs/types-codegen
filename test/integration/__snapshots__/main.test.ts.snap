// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`command > include all routes 1`] = `
"// Generated by @duplojs/types-codegen
/* eslint-disable */
/* prettier-ignore */
/* istanbul ignore file */
/* v8 ignore start */
// noinspection JSUnusedGlobalSymbols
// @ts-nocheck
interface CodegenReceiveFormData<GenericValue extends Record<string, string | string[] | number | number[] | Date | Date[] | File[]>> {
    extractor: (...args: any[]) => Promise<GenericValue>;
}

export { CodegenReceiveFormData };

type CodegenRoutes = ({
    method: "POST";
    path: "/pictures";
    body: CodegenReceiveFormData<{
        userId: string;
        pictures: File[];
    }>;
    headers?: {
        "test-headers"?: string | undefined;
    } | undefined;
    response: {
        code: 200;
        information: "picture.upload";
        body: File;
    };
}) | ({
    method: "GET";
    path: "/users/{userId}";
    params: {
        userId: string;
    };
    response: {
        code: 404;
        information: "user.notfound";
        body?: undefined;
    } | {
        code: 200;
        information: "user.get";
        body: {
            name: string;
            age: number;
        };
    };
}) | ({
    method: "POST";
    path: "/users";
    body: {
        name: string;
        age: number;
    };
    response: {
        code: 201;
        information: "user.created";
        body: {
            name: string;
            age: number;
        };
    };
}) | ({
    method: "GET";
    path: "/users";
    response: {
        code: 200;
        information: "users.get";
        body: {
            name: string;
            age: number;
        }[];
    };
});

export { CodegenRoutes };
/* v8 ignore stop */
"
`;

exports[`command > require scripts 1`] = `
"// Generated by @duplojs/types-codegen
/* eslint-disable */
/* prettier-ignore */
/* istanbul ignore file */
/* v8 ignore start */
// noinspection JSUnusedGlobalSymbols
// @ts-nocheck
interface CodegenReceiveFormData<GenericValue extends Record<string, string | string[] | number | number[] | Date | Date[] | File[]>> {
    extractor: (...args: any[]) => Promise<GenericValue>;
}

export { CodegenReceiveFormData };

type CodegenRoutes = ({
    method: "GET";
    path: "/users/{userId}";
    params: {
        userId: string;
    };
    response: {
        code: 404;
        information: "user.notfound";
        body?: undefined;
    } | {
        code: 200;
        information: "user.get";
        body: {
            name: string;
            age: number;
        };
    };
}) | ({
    method: "POST";
    path: "/users";
    body: {
        name: string;
        age: number;
    };
    response: {
        code: 201;
        information: "user.created";
        body: {
            name: string;
            age: number;
        };
    };
}) | ({
    method: "GET";
    path: "/users";
    response: {
        code: 200;
        information: "users.get";
        body: {
            name: string;
            age: number;
        }[];
    };
}) | ({
    method: "POST";
    path: "/pictures";
    body: CodegenReceiveFormData<{
        userId: string;
        pictures: File[];
    }>;
    headers?: {
        "test-headers"?: string | undefined;
    } | undefined;
    response: {
        code: 200;
        information: "picture.upload";
        body: File;
    };
});

export { CodegenRoutes };
/* v8 ignore stop */
"
`;
